File: Shared/Exceptions/BaseException.cs
﻿using Shared.Responses;

namespace Shared.Exceptions
{
public class BaseException : Exception
{
    public enum EType : byte
    {
        Unknown,
        FromTo,
        Value,
        Database
    }

    public ErrorResponse Error { get; }
    public EType Type { get; }

    protected BaseException(EType type, ErrorResponse error) : base(error.Message)
    {
        Type = type;

        Error = error;
    }
}
}

File: Web/src/models/exceptions/BaseException.ts
import { ToExceptionStr } from "../../utilities/EnumsExtensions";
import { ErrorResponse } from "../responses/ErrorResponse";

export enum EException {
  Unknown,
  FromTo,
  Value,
  Database,
}

export class BaseException extends Error {
  public error: ErrorResponse;
  public type: EException = EException.Unknown;

  protected constructor(type: EException, error: ErrorResponse) {
    super(error.message);
    super.name = ToExceptionStr(type);

    this.type = type;
    this.error = error;
    this.error.typeException = type;
  }
}
